### ISSUE TITLE
s390x: CI: panic: DockerSuite.TestHealth test timed out

### ISSUE THREAD

**Issue by michael-holzheu** [docker/docker#26595]
On s390x the CI test DockerSuite.TestHealth fails in-deterministically with timeout:

```
...
PASS: docker_api_containers_test.go:321: DockerSuite.TestGetStoppedContainerStats       0.273s
PASS: docker_api_version_test.go:13: DockerSuite.TestGetVersion 0.001s
--- FAIL: Test (11638.56s)
panic: DockerSuite.TestHealth test timed out after 3h0m0s [recovered]
        panic: DockerSuite.TestHealth test timed out after 3h0m0s

goroutine 9 [running]:
panic(0x80e23120, 0xc4216a6010)
        /usr/local/go/src/runtime/panic.go:500 +0x408
testing.tRunner.func1(0xc420070a80)
        /usr/local/go/src/testing/testing.go:579 +0x1a2
panic(0x80e23120, 0xc4216a6010)
        /usr/local/go/src/runtime/panic.go:458 +0x53c
github.com/go-check/check.(*suiteRunner).runTest(0xc420079c20, 0xc420411e30, 0xc4201cdb30)
        /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:875 +0x264
github.com/go-check/check.(*suiteRunner).run(0xc420079c20, 0x814e0480)
        /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:621 +0x2bc
github.com/go-check/check.Run(0x80efb820, 0x814e0480, 0xc4203d4cd0, 0x7fffff)
        /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:100 +0x54
github.com/go-check/check.RunAll(0xc4203d4cd0, 0x0)
        /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:92 +0xee
github.com/go-check/check.TestingT(0xc420070a80)
        /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/run.go:80 +0x2ea
github.com/docker/docker/integration-cli.Test(0xc420070a80)
        /go/src/github.com/docker/docker/integration-cli/check_test.go:29 +0x134
testing.tRunner(0xc420070a80, 0x8101af60)
        /usr/local/go/src/testing/testing.go:610 +0xd8
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:646 +0x376

goroutine 1 [chan receive, 193 minutes]:
testing.(*T).Run(0xc420070a80, 0x80f00e76, 0x4, 0x8101af60, 0x80032c32)
        /usr/local/go/src/testing/testing.go:647 +0x3a0
testing.RunTests.func1(0xc4200709c0)
        /usr/local/go/src/testing/testing.go:793 +0xe6
testing.tRunner(0xc4200709c0, 0xc420043da8)
        /usr/local/go/src/testing/testing.go:610 +0xd8
testing.RunTests(0x8101db50, 0x81473f70, 0x1, 0x1, 0x100000080dc85c0)
        /usr/local/go/src/testing/testing.go:799 +0x3da
testing.(*M).Run(0xc420043ef8, 0xc420379ff0)
        /usr/local/go/src/testing/testing.go:743 +0x88
main.main()
        github.com/docker/docker/integration-cli/_test/_testmain.go:54 +0x14c

goroutine 17 [syscall, 193 minutes, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_s390x.s:1021 +0x2

goroutine 5 [syscall, 193 minutes]:
os/signal.signal_recv(0x0)
        /usr/local/go/src/runtime/sigqueue.go:116 +0x1be
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:22 +0x1e
created by os/signal.init.1
        /usr/local/go/src/os/signal/signal_unix.go:28 +0x3c

goroutine 9086 [syscall, 180 minutes]:
syscall.Syscall6(0x119, 0x1, 0x7d3a, 0xc420674880, 0x1000004, 0x0, 0x0, 0x0, 0xc420384c00, 0xc421b0f2c0)
        /usr/local/go/src/syscall/asm_linux_s390x.s:39 +0x12
os.(*Process).blockUntilWaitable(0xc420495d10, 0x80eb3ac0, 0x0, 0x0)
        /usr/local/go/src/os/wait_waitid.go:28 +0x96
os.(*Process).wait(0xc420495d10, 0x0, 0x0, 0x0)
        /usr/local/go/src/os/exec_unix.go:22 +0x96
os.(*Process).Wait(0xc420495d10, 0xc421b251e0, 0x0, 0x0)
        /usr/local/go/src/os/doc.go:49 +0x30
os/exec.(*Cmd).Wait(0xc4201f0c60, 0x0, 0x0)
        /usr/local/go/src/os/exec/exec.go:434 +0x1a4
github.com/docker/docker/pkg/integration/cmd.WaitOnCmd(0x0, 0xc421b6d300, 0x4)
        /go/src/github.com/docker/docker/pkg/integration/cmd/command.go:281 +0x42
github.com/docker/docker/pkg/integration/cmd.RunCmd(0xc421b6d280, 0x4, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/docker/docker/pkg/integration/cmd/command.go:232 +0x6a
github.com/docker/docker/pkg/integration/cmd.RunCommand(0xc42021ba90, 0x48, 0xc4228d75d8, 0x3, 0x3, 0x2)
        /go/src/github.com/docker/docker/pkg/integration/cmd/command.go:237 +0x1dc
github.com/docker/docker/integration-cli.dockerCmd(0xc421c662d0, 0xc4228d75d8, 0x3, 0x3, 0x0, 0x0, 0x1)
        /go/src/github.com/docker/docker/integration-cli/docker_utils.go:468 +0xf2
github.com/docker/docker/integration-cli.(*DockerSuite).TestHealth(0x814e0480, 0xc421c662d0)
        /go/src/github.com/docker/docker/integration-cli/docker_cli_health_test.go:151 +0x1d5a
reflect.Value.call(0x80efb820, 0x814e0480, 0x82213, 0x80f00bb6, 0x4, 0xc4228d7f00, 0x1, 0x1, 0x0, 0x0, ...)
        /usr/local/go/src/reflect/value.go:434 +0xf28
reflect.Value.Call(0x80efb820, 0x814e0480, 0x82213, 0xc4228d7f00, 0x1, 0x1, 0x0, 0x0, 0x0)
        /usr/local/go/src/reflect/value.go:302 +0xa8
github.com/go-check/check.(*suiteRunner).forkTest.func1(0xc421c662d0)
        /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:816 +0x400
github.com/go-check/check.(*suiteRunner).forkCall.func1(0xc420079c20, 0xc421c662d0, 0xc421aebfc0)
        /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:672 +0xa2
created by github.com/go-check/check.(*suiteRunner).forkCall
        /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:673 +0x2da

goroutine 21 [runnable]:
github.com/go-check/check.(*resultTracker)._loopRoutine(0xc420079b90)
        /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:461 +0x26e
created by github.com/go-check/check.(*resultTracker).start
        /go/src/github.com/docker/docker/vendor/src/github.com/go-check/check/check.go:441 +0x3c

goroutine 9334 [syscall, 180 minutes]:
syscall.Syscall(0x3, 0x7, 0xc4205dfc00, 0x200, 0x0, 0x3ff90035000, 0x0)
        /usr/local/go/src/syscall/asm_linux_s390x.s:13 +0x12
syscall.read(0x7, 0xc4205dfc00, 0x200, 0x200, 0x80034000, 0x0, 0x0)
        /usr/local/go/src/syscall/zsyscall_linux_s390x.go:781 +0x68
syscall.Read(0x7, 0xc4205dfc00, 0x200, 0x200, 0x0, 0x0, 0x0)
        /usr/local/go/src/syscall/syscall_unix.go:161 +0x54
os.(*File).read(0xc420253f58, 0xc4205dfc00, 0x200, 0x200, 0xc4205dfc00, 0x0, 0x0)
        /usr/local/go/src/os/file_unix.go:228 +0x60
os.(*File).Read(0xc420253f58, 0xc4205dfc00, 0x200, 0x200, 0x200000003, 0x0, 0x0)
        /usr/local/go/src/os/file.go:101 +0x86
bytes.(*Buffer).ReadFrom(0xc4202fa770, 0x814854e0, 0xc420253f58, 0x0, 0x0, 0x0)
        /usr/local/go/src/bytes/buffer.go:176 +0x262
io.copyBuffer(0x81483420, 0xc4202fa770, 0x814854e0, 0xc420253f58, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/io/io.go:384 +0x158
io.Copy(0x81483420, 0xc4202fa770, 0x814854e0, 0xc420253f58, 0x0, 0x0, 0x0)
        /usr/local/go/src/io/io.go:360 +0x5a
os/exec.(*Cmd).writerDescriptor.func1(0x0, 0x0)
        /usr/local/go/src/os/exec/exec.go:253 +0x7a
os/exec.(*Cmd).Start.func1(0xc4201f0c60, 0xc421b251c0)
        /usr/local/go/src/os/exec/exec.go:370 +0x26
created by os/exec.(*Cmd).Start
        /usr/local/go/src/os/exec/exec.go:371 +0x8f2

goroutine 9333 [syscall, 180 minutes]:
syscall.Syscall(0x3, 0x5, 0xc4205dfe00, 0x200, 0x0, 0x3ff90035000, 0x0)
        /usr/local/go/src/syscall/asm_linux_s390x.s:13 +0x12
syscall.read(0x5, 0xc4205dfe00, 0x200, 0x200, 0x80034000, 0x0, 0x0)
        /usr/local/go/src/syscall/zsyscall_linux_s390x.go:781 +0x68
syscall.Read(0x5, 0xc4205dfe00, 0x200, 0x200, 0x0, 0x0, 0x0)
        /usr/local/go/src/syscall/syscall_unix.go:161 +0x54
os.(*File).read(0xc420253f40, 0xc4205dfe00, 0x200, 0x200, 0xc4205dfe00, 0x0, 0x0)
        /usr/local/go/src/os/file_unix.go:228 +0x60
os.(*File).Read(0xc420253f40, 0xc4205dfe00, 0x200, 0x200, 0x200000003, 0x0, 0x0)
        /usr/local/go/src/os/file.go:101 +0x86
bytes.(*Buffer).ReadFrom(0xc4202fa700, 0x814854e0, 0xc420253f40, 0x0, 0x0, 0x0)
        /usr/local/go/src/bytes/buffer.go:176 +0x262
io.copyBuffer(0x81483420, 0xc4202fa700, 0x814854e0, 0xc420253f40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/io/io.go:384 +0x158
io.Copy(0x81483420, 0xc4202fa700, 0x814854e0, 0xc420253f40, 0x1cb, 0x0, 0x0)
        /usr/local/go/src/io/io.go:360 +0x5a
os/exec.(*Cmd).writerDescriptor.func1(0x0, 0x0)
        /usr/local/go/src/os/exec/exec.go:253 +0x7a
os/exec.(*Cmd).Start.func1(0xc4201f0c60, 0xc421b25140)
        /usr/local/go/src/os/exec/exec.go:370 +0x26
created by os/exec.(*Cmd).Start
        /usr/local/go/src/os/exec/exec.go:371 +0x8f2
---> Making bundle: .integration-daemon-stop (in bundles/1.13.0-dev/test-integration-cli)
```

The first failure was detected in Jenkins run 524: https://jenkins.dockerproject.org/job/Docker%20Master%20(z)/524/

This run was for commit 19f0feadd6405176f7cbe94e0ad4b9dc1965da3e. But I assume the root cause was some commit before.

**Comment by michael-holzheu** [comment]
Investigating...

**Comment by runcom** [comment]
@michael-holzheu probably a duplicate of https://github.com/docker/docker/issues/26506 (?)

**Comment by michael-holzheu** [comment]
> @michael-holzheu probably a duplicate of #26506 (?)

Perhaps, I try to reproduce and get the docker.log

**Comment by thaJeztah** [comment]
@michael-holzheu have you been able to check if it's a duplicate?

**Comment by michael-holzheu** [comment]
> @michael-holzheu have you been able to check if it's a duplicate?

I tried, but up to now I was not successful to reproduce manually...

**Comment by LK4D4** [comment]
I have a fix included in #26538, I'll run z there.

**Comment by michael-holzheu** [comment]
> I have a fix included in #26538, I'll run z there.

@LK4D4 : Looks like this fixed also our problem on z Systems: 

See the following jenkins run:

 * https://jenkins.dockerproject.org/job/Docker%20Master%20(z)/623/

The last 5 runs have been successful:

 * https://jenkins.dockerproject.org/job/Docker%20Master%20(z)/

**Comment by LK4D4** [comment]
Cool!
